filters/flowid/standard.go

```diff
@@ -0,0 +1,76 @@
+package flowid
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"regexp"
+)
+
+const (
+	flowIdAlphabet  = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-+"
+	alphabetBitMask = 63
+	MaxLength       = 64
+	MinLength       = 8
+	defaultLen      = 16
+)
+
+var (
+	ErrInvalidLen       = errors.New(fmt.Sprintf("Invalid length. Must be between %d and %d", MinLength, MaxLength))
+	standardFlowIDRegex = regexp.MustCompile(`^[0-9a-zA-Z+-]+$`)
+)
+
+// NewFlowId creates a new standard generator with the defined length and returns a flowid
+// This exported function is deprecated and the new Generator interface should be used
+func NewFlowId(l int) (string, error) {
+	g, err := NewStandardGenerator(l)
+	if err != nil {
+		return "", fmt.Errorf("deprecated new flowid: %v", err)
+	}
+	return g.Generate()
+}
+
+type standardGenerator struct {
+	length int
+}
+
+// NewStandardGenerator creates a new FlowID generator that generates flow IDs with length l
+func NewStandardGenerator(l int) (Generator, error) {
+	if l < MinLength || l > MaxLength {
+		return nil, ErrInvalidLen
+	}
+
+	return &standardGenerator{length: l}, nil
+}
+
+// Generate returns a new Flow ID from the built-in generator with the configured length
+// The alphabet is limited to 64 elements and requires a random 6 bit value to index any of them.
+// The cost to rnd.IntXX is not very relevant but the bit shifting operations are faster.
+// For this reason a single call to rnd.Int63 is used and its bits are mapped up to 10 chunks of 6 bits each.
+// The byte data type carries 2 additional bits for the next chunk which are cleared with the alphabet bit mask.
```

Comment:
your're aware that comments of the unpublished members, like standardGenerator won't be visible in godoc? This doc contains useful information. Please move the gist of it to the NewStandardGenerator, as somthing like NewStandardGenerator creates a new FlowID generator that ... <this content>